tfexec/version.go

```diff
@@ -180,6 +181,22 @@ func (tf *Terraform) compatible(ctx context.Context, minInclusive *version.Versi
 	return nil
 }
 
+// experimentsEnabled asserts the cached terraform version has experiments enabled in the executable,
+// and returns a well known error if not. Experiments are enabled in alpha and (potentially) dev builds of Terraform.
+func (tf *Terraform) experimentsEnabled(ctx context.Context) error {
```

Comment:
I would've probably called this `versionIsPrerelease()` or something along these lines, as that's what the function actually does but it's also not exposed API to downstream so I'm not too worried about the name here.